src/shims/foreign_items/posix.rs

```diff
@@ -313,11 +310,53 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_null(dest)?;
             }
 
-            // Incomplete shims that we "stub out" just to get pre-main initialization code to work.
+            // Threading
+            "pthread_create" => {
+                assert_eq!(args.len(), 4);
+                let result = this.pthread_create(args[0], args[1], args[2], args[3])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_join" => {
+                assert_eq!(args.len(), 2);
+                let result = this.pthread_join(args[0], args[1])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_detach" => {
+                assert_eq!(args.len(), 1);
+                let result = this.pthread_detach(args[0])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_self" => {
+                assert_eq!(args.len(), 0);
+                this.pthread_self(dest)?;
+            }
+            "prctl" => {
+                assert_eq!(args.len(), 5);
+                let result = this.prctl(args[0], args[1], args[2], args[3], args[4])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+
+            "pthread_attr_getguardsize" => {
```

Comment:
This uses our fake `PAGE_SIZE`, so can we really call this a proper shim? Or should it be moved down to the incomplete shims and only permit libstd to call it?

Reply:
We do not provide `pthread_attr_setguardsize` yet, so we can return whatever we want as long as it is [more than 0](https://github.com/rust-lang/rust/blob/ce1ab355c27faa84f984b15800cd6a65e674b9f2/src/libstd/sys/unix/thread.rs#L405-L406). However, since we do not implement any other of `pthread_attr` shims, I have moved this one below near the incomplete shims.